### PR TITLE
feat(FX-3997): Expand New 'Save' Styling to ShelfArtwork component

### PR DESCRIPTION
**Pull request by nickskalkin**
Co-authored-by: Dima Tretyak (Dzmitry Tratsiak) <itretyak@ya.ru>

The type of this PR is: **feature**

This PR solves [FX-3997](https://artsyproduct.atlassian.net/browse/FX-3997)

Review app: https://shelf-artwork-save-button.artsy.net/

## Description

Updates the pages which do not have the quick-view functionality to have the new Save button.

### Before

https://user-images.githubusercontent.com/3934579/172379667-91c3b3c5-f479-4681-9991-ad13f7a59d89.mp4

### After 

https://user-images.githubusercontent.com/3934579/172379684-74d499b1-5bf7-48a1-9bef-870d7108c424.mp4

### CODE REVIEW

**Review by dimatretyak — APPROVED** [review]

File: src/v2/Components/Artwork/ShelfArtwork.tsx [hunk 1]
```diff
@@ -43,17 +44,24 @@ const ShelfArtwork: React.FC<ShelfArtworkProps> = ({
...
+  const shouldShowHoverSaveButton =
+    isHoverEffectEnabled && (!!artwork.is_saved || isSaveButtonVisible)
 
   return (
-    <>
+    <div
+      {...containerProps}
+      data-test="artworkShelfArtwork"
+      data-testid="ShelfArtwork"
+    >
```
Comment (dzucconi) [inline src/v2/Components/Artwork/ShelfArtwork.tsx:59]: Why do we need an extraneous wrapping `div`?

**Comment by dzucconi** [comment]
Really happy to see this piece of UI finally updated

**Comment by dzucconi** [comment]
Though wondering why it's needed to feature flag such a minor change?

**Comment by dimatretyak** [comment]
> Though wondering why it's needed to feature flag such a minor change?

[GridItem](https://github.com/artsy/force/blob/main/src/v2/Components/Artwork/GridItem.tsx#L39) component still uses the feature flag and for the reason I decided to use it here. Also in the near future this flag will be removed and we will need to make a small code refactor (remove unused old code) and it seems to me that it is better to do it all in a separate PR as a single task

File: src/v2/Components/Artwork/ShelfArtwork.tsx [hunk 2]
```diff
@@ -84,9 +94,10 @@ const ShelfArtwork: React.FC<ShelfArtworkProps> = ({
           hideArtistName={hideArtistName}
           hideSaleInfo={hideSaleInfo}
           maxWidth={artwork.image?.resized?.width}
+          shouldShowHoverSaveButton={!!shouldShowHoverSaveButton}
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/v2/Components/Artwork/ShelfArtwork.tsx:97]: I will change this name as part of the feature flag removal task

File: src/v2/Components/Artwork/ShelfArtwork.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/v2/Components/Artwork/ShelfArtwork.tsx:59]: We need to show the "save" button when the user hovered over the `ShelfArtwork` component, and not on the image (how it is now done on production). For this reason, we wrapped the contents of the component in a `div` and pass the `onMouseEnter` and `onMouseLeave` handlers

Comment (dzucconi) [reply src/v2/Components/Artwork/ShelfArtwork.tsx:59]: Oh, I see — didn't realize the link didn't wrap the whole thing 👍

**Review by dzucconi — APPROVED** [review]

File: src/v2/Components/Artwork/ShelfArtwork.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/v2/Components/Artwork/ShelfArtwork.tsx:97]: Changed in this PR #10298

**Pull request merged**